src/styles/utils/function.scss

```diff
@@ -0,0 +1,30 @@
+@import "variables.scss";
+
+@for $index from 1 through 31 {
+  .calendar-day:nth-child(#{$index})::before {
+    content: "#{$index}";
+  }
+}
+
+$weekdays:
+  "mon" 0,
+  "tue" #{$day-margin},
+  "wed" calc(2 * #{$day-margin}),
+  "thu" calc(3 * #{$day-margin}),
+  "fri" calc(4 * #{$day-margin}),
+  "sat" calc(5 * #{$day-margin}),
+  "sun" calc(6 * #{$day-margin});
+
+@each $weekday, $shift in $weekdays {
+  .calendar-day--start-day_#{$weekday} {
+    margin-left: $shift;
+  }
+}
+
+@for $index from 28 through 31 {
+  .calendar--month-length_#{$index} {
+    .calendar-day:nth-child(n + #{$index+1}) {
+      display: none;
+    }
+  }
+}
```

Comment:
calendar block + nesting

Reply:
these are functions too. Why should they be in a calendar block?